.pylintrc

```diff
@@ -1,7 +1,49 @@
 [MASTER]
-disable=missing-function-docstring,
+# This is a workaround for pylint not being able to trace our changing of the
+# import path in tests so we can find the actual module.
+init-hook='import sys; sys.path.append(".")'
+
+disable=
+  consider-using-f-string,
   missing-class-docstring,
-  too-many-arguments, too-many-branches,
+  missing-function-docstring,
+  too-many-arguments,
+  too-many-branches,
   too-many-instance-attributes,
-  too-many-lines, too-many-statements,
-  consider-using-f-string
+  too-many-lines,
+  too-many-statements,
+
+  # We previously ignored all these non-error lints.
+  #
+  # TODO(bbannier): audit this list and fix lints where possible.
+  anomalous-backslash-in-string,
+  broad-exception-caught,
+  consider-using-dict-items,
+  consider-using-with,
+  cyclic-import,
+  duplicate-code,
+  expression-not-assigned,
+  fixme,
+  implicit-str-concat,
+  invalid-name,
+  line-too-long,
+  missing-module-docstring,
+  pointless-statement,
+  protected-access,
+  redefined-builtin,
+  redefined-outer-name,
+  too-few-public-methods,
+  too-many-locals,
+  too-many-public-methods,
+  too-many-return-statements,
+  unidiomatic-typecheck,
+  unnecessary-lambda-assignment,
+  unspecified-encoding,
+  unused-argument,
+  unused-import,
+  unused-variable,
```

Comment:
Seems `unused-import` and `unused-variable` would be good to fixup as other tooling likely puts wiggles under those, too.

Some of the others seem scary, but they are also in test code.

Reply:
Totally agree that we should look into these which is also exactly why I put them all of these a separate block (e.g., redef'd builtins confuse some analyses, cyclic dependencies lead to unresolved types with pylint, ...)

What this patch does is make sure we do not introduce new lint categories, and have a way forward to tackle these (one by one, or in one giant sweep). TBH, I would like to merge this as is, and then we can have a look at cleaning up existing code later.